From 920956ddf41b3b2304c02ba19f4f203d5cf67e1a Mon Sep 17 00:00:00 2001
From: Roman Beranek <roman.beranek@prusa3d.com>
Date: Mon, 12 Jul 2021 15:44:37 +0200
Subject: [PATCH] env: reset defaults on load

---
 env/Kconfig   |  7 +++++++
 env/common.c  |  3 +++
 2 files changed, 10 insertions(+)

diff --git a/env/Kconfig b/env/Kconfig
index ed12609f6a..6a8d21ad75 100644
--- a/env/Kconfig
+++ b/env/Kconfig
@@ -581,6 +581,13 @@ config ENV_VARS_UBOOT_RUNTIME_CONFIG
 	  run-time determined information about the hardware to the
 	  environment.  These will be named board_name, board_rev.
 
+config ENV_RESET_DEFAULTS_ON_LOAD
+	bool "Keep environment fresh"
+	help
+	  Once an environment is loaded from the eMMC, reset it to
+	  defaults. This will keep variables without default values
+	  as they were.
+
 if SPL_ENV_SUPPORT
 config SPL_ENV_IS_NOWHERE
 	bool "SPL Environment is not stored"
diff --git a/env/common.c b/env/common.c
index 1fd1bd01d3..1b50f6c375 100644
--- a/env/common.c
+++ b/env/common.c
@@ -240,6 +240,9 @@ void env_relocate(void)
 #endif
 	} else {
 		env_load();
+#if defined(CONFIG_ENV_RESET_DEFAULTS_ON_LOAD) && !defined(CONFIG_SPL_BUILD)
+		env_set_default_vars(0, NULL, 0);
+#endif
 	}
 }
 
-- 
2.32.0
