From 8e009d82e7e2c23306edf4bd33b9e32f1dc885b1 Mon Sep 17 00:00:00 2001
From: Roman Beranek <roman.beranek@prusa3d.com>
Date: Fri, 12 Apr 2019 17:29:49 +0200

---
 board/sunxi/board.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/board/sunxi/board.c b/board/sunxi/board.c
index 3ca6b62831..0ddb96cd5b 100644
--- a/board/sunxi/board.c
+++ b/board/sunxi/board.c
@@ -573,6 +573,11 @@ int board_mmc_init(bd_t *bis)
 {
 	__maybe_unused struct mmc *mmc0, *mmc1;
 
+        // TODO:
+        // HACK:
+        // This is a hack for MMC read problems on prusa-sl1 A64 board
+        mdelay(1500);
+
 	mmc_pinmux_setup(CONFIG_MMC_SUNXI_SLOT);
 	mmc0 = sunxi_mmc_init(CONFIG_MMC_SUNXI_SLOT);
 	if (!mmc0)
