From 036971fe1834723baaae7aca2e34e9b5e01c5a83 Mon Sep 17 00:00:00 2001
From: Roman Beranek <roman.beranek@prusa3d.com>
Date: Fri, 18 Sep 2020 12:26:22 +0000
Subject: [PATCH 10/12] concat-bootargs-keep-extra

---
 include/configs/sunxi-common.h | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index f052c0296a..a97fb566e4 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -485,16 +485,17 @@ extern int soft_i2c_gpio_scl;
 	"\0"
 
 #define PRUSA_KERNEL_CMDLINE \
-	"console=${console} root=${rootdev} rootwait " \
-	"panic=10 video=HDMI-A-1:D drm.edid_firmware=HDMI-A-1:edid/ls055r1sx04_148.5mhz.bin " \
-	"logo.nologo vt.global_cursor_default=0 board_revision=${board_revision} stmmac.quirks=${eth_quirks} ${extra}"
+	"rootwait panic=10 video=HDMI-A-1:D drm.edid_firmware=HDMI-A-1:edid/ls055r1sx04_148.5mhz.bin logo.nologo " \
+	"vt.global_cursor_default=0 console=${console} board_revision=${board_revision} stmmac.quirks=${eth_quirks}"
 
 
 #define BOOTCMD_PRUSA \
 	"bootcmd_prusa=" \
+		"setenv bootargs " PRUSA_KERNEL_CMDLINE "; " \
+		"setenv bootargs ${bootargs} ${extra}; " \
+		"setenv extra; " \
 		"if test ${mmc_bootdev} -eq 0; then " \
-			"setenv extra; " \
-			"setenv rootdev /dev/mmcblk0p1; " \
+			"setenv bootargs ${bootargs} root=/dev/mmcblk0p1; " \
 			"setenv mmc_bootpart 1; " \
 			"if test -e mmc 0:1 boot.scr; then " \
 				"echo Found U-Boot script; " \
@@ -503,12 +504,11 @@ extern int soft_i2c_gpio_scl;
 			"fi; " \
 		"else " \
 			"run update_active_rootfs_slot; " \
-			"setenv extra rauc.slot=${slot}; " \
-			"setenv rootdev PARTLABEL=${rootlabel}; " \
+			"setenv bootargs ${bootargs} rauc.slot=${slot}; " \
+			"setenv bootargs ${bootargs} root=PARTLABEL=${rootlabel}; " \
 			"gpt part-num mmc 1 ${rootlabel} mmc_bootpart; " \
 		"fi; " \
 		"run scan_dev_for_fdt; " \
-		"setenv bootargs " PRUSA_KERNEL_CMDLINE "; " \
 		"" \
 		"if load mmc ${mmc_bootdev}:${mmc_bootpart} ${kernel_addr_r} ${boot_prefix}Image; " \
 		"then " \
-- 
2.28.0

