diff --git a/board/sunxi/board.c b/board/sunxi/board.c
index 98bc3cd0c1..c764c9bc7a 100644
--- a/board/sunxi/board.c
+++ b/board/sunxi/board.c
@@ -570,6 +570,11 @@ int board_mmc_init(bd_t *bis)
 {
 	__maybe_unused struct mmc *mmc0, *mmc1;
 
+        // TODO:
+        // HACK:
+        // This is a hack for MMC read problems on prusa-sl1 A64 board
+        mdelay(1500);
+
 	mmc_pinmux_setup(CONFIG_MMC_SUNXI_SLOT);
 	mmc0 = sunxi_mmc_init(CONFIG_MMC_SUNXI_SLOT);
 	if (!mmc0)
