diff --git a/src/emmc.c b/src/emmc.c
index 83fec96..7d7c536 100644
--- a/src/emmc.c
+++ b/src/emmc.c
@@ -46,6 +46,12 @@ gboolean r_emmc_read_bootpart(const gchar *device, gint *bootpart_active, GError
 	}
 
 	ret = r_emmc_read_extcsd(fd, extcsd);
+	if (ret) {
+		/* wait */
+		sleep(1);
+		/* try again */
+		ret = r_emmc_read_extcsd(fd, extcsd);
+	}
 	if (ret) {
 		g_set_error(error, R_UPDATE_ERROR, R_UPDATE_ERROR_FAILED,
 				"Could not read from extcsd register %d in %s\n",
@@ -106,6 +112,12 @@ gboolean r_emmc_write_bootpart(const gchar *device, gint bootpart_active, GError
 		value = 0x10;
 
 	ret = r_emmc_write_extcsd(fd, EXT_CSD_PART_CONFIG, value);
+	if (ret) {
+                /* wait */
+                sleep(1);
+                /* try again */
+                ret = r_emmc_write_extcsd(fd, EXT_CSD_PART_CONFIG, value);
+        }
 	if (ret) {
 		g_set_error(error, R_UPDATE_ERROR, R_UPDATE_ERROR_FAILED,
 				"Could not write 0x%02x to extcsd register %d in %s\n",
