diff --git a/firmware/home/root/sl1fw/defines.py b/firmware/home/root/sl1fw/defines.py
index 2021e98..96e4e98 100644
--- a/firmware/home/root/sl1fw/defines.py
+++ b/firmware/home/root/sl1fw/defines.py
@@ -14,7 +14,7 @@ usbPath = os.path.join(home, "usb")
 ramdiskPath = os.path.join(home, "ramdisk")
 printerlog = os.path.join(ramdiskPath, "sl1fw.log")
 jobCounter = os.path.join(home, "jobs.log")
-hwConfigFile = os.path.join(swPath, "hardware.cfg")
+hwConfigFile = "/etc/sl1fw/hardware.cfg"
 
 configFile = "config.ini"
 
