--- a/sl1fw/defines.py  2018-11-14 14:57:08.639228447 +0100
+++ b/sl1fw/defines.py      2018-11-15 16:23:33.498706290 +0100
@@ -11,7 +11,7 @@
 swPath = os.path.join(home, "sl1fw")
 usbPath = os.path.join(home, "usb")
 #usbPath = os.path.join(home, "project")
-ramdiskPath = os.path.join(home, "ramdisk")
+ramdiskPath = "/run/sl1fw"
 printerlog = os.path.join(ramdiskPath, "sl1fw.log")
 jobCounter = os.path.join(home, "jobs.log")
 hwConfigFile = os.path.join(swPath, "hardware.cfg")
 
