--- /dev/null	2018-11-22 14:04:10.685533267 +0100
+++ a/setup.py	2018-11-26 12:34:59.576174620 +0100
@@ -0,0 +1,16 @@
+from setuptools import setup, find_packages
+from glob import glob
+
+setup(
+    name="sl1fw",
+    version="0.1",
+    packages=find_packages(),
+    scripts=['sl1fw/main.py'],
+    package_data={'sl1fw': ['data/*']},
+    data_files=[('scripts', glob('sl1fw/scripts/*')),
+                ('/etc/sl1fw', ['sl1fw/hardware.cfg']),
+                ('/srv/http/intranet', glob('sl1fw/intranet/*.html')),
+                ('/srv/http/intranet/templates', glob('sl1fw/intranet/templates/*.html')),
+                ('/srv/http/intranet/static', glob('sl1fw/intranet/static/*'))]
+)
+
