diff --git a/main.qml b/main.qml
index e3e59bb..5703bd3 100644
--- a/main.qml
+++ b/main.qml
@@ -69,8 +69,7 @@ Window {
                     if("page" in pMsg) {
                         console.log("showPage: " + pMsg["page"])
                         if(pMsg["page"] === "home" ) {
-                            view.clear()
-                            view.push(homeScreen);
+                            view.replace(homeScreen);
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "keyboard_edit") {
@@ -102,16 +101,14 @@ Window {
                         else if(pMsg["page"] === "sourceselect") {
                             txtSelectProjectSourceLabel.text = pMsg["line1"]
                             txtSelectProjectSourceIp.text = pMsg["line2"]
-                            view.clear()
-                            view.push(printSourceSelectScreen);
+                            view.replace(printSourceSelectScreen);
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "confirm" ) {
                             if("line1" in pMsg) confirmScreenLine1.text = pMsg["line1"]; else confirmScreenLine1.text = ""
                             if("line2" in pMsg) confirmScreenLine2.text = pMsg["line2"]; else confirmScreenLine2.text = ""
                             if("line3" in pMsg) confirmScreenLine3.text = pMsg["line3"]; else confirmScreenLine3.text = ""
-                            view.clear()
-                            view.push(confirmScreen)
+                            view.replace(confirmScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "netinfo") {
@@ -122,14 +119,12 @@ Window {
                             if("qr2" in pMsg) netinfoQRCode2.value = pMsg["qr2"]; else netinfoQRCode2.value = ""
                             if("qr1label" in pMsg) netinfoScreenQr1Label.text = pMsg["qr1label"]; else netinfoScreenQr1Label.text = ""
                             if("qr2label" in pMsg) netinfoScreenQr2Label.text = pMsg["qr2label"]; else netinfoScreenQr2Label.text = ""
-                            view.clear()
-                            view.push(netinfoScreen);
+                            view.replace(netinfoScreen);
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "control") {
                             console.log("page: control")
-                            view.clear()
-                            view.push(controlScreen);
+                            view.replace(controlScreen);
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "wait") {
@@ -137,8 +132,7 @@ Window {
                             if("line1" in pMsg) waitScreenLine1.text = pMsg["line1"]; else waitScreenLine1.text = ""
                             if("line2" in pMsg) waitScreenLine2.text = pMsg["line2"]; else waitScreenLine2.text = ""
                             if("line3" in pMsg) waitScreenLine3.text = pMsg["line3"]; else waitScreenLine3.text = ""
-                            view.clear()
-                            view.push(waitScreen);
+                            view.replace(waitScreen);
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "about") {
@@ -147,8 +141,7 @@ Window {
                             if("line3" in pMsg) aboutScreenLine3.text = pMsg["line3"]; else aboutScreenLine2.text = ""
                             if("qr1" in pMsg) aboutQr.value = pMsg["qr1"]; else aboutQr.value = ""
 
-                            view.clear()
-                            view.push(aboutScreen)
+                            view.replace(aboutScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "admin") {
@@ -167,8 +160,7 @@ Window {
                             if("button13" in pMsg) btnAdmin13.text = pMsg["button13"]; else btnAdmin13.text = ""
                             if("button14" in pMsg) btnAdmin14.text = pMsg["button14"]; else btnAdmin14.text = ""
                             //if("button15" in pMsg) btnAdmin15.text = pMsg["button15"]; else btnAdmin15.text = ""
-                            view.clear()
-                            view.push(adminScreen)
+                            view.replace(adminScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "setup") {
@@ -217,30 +209,26 @@ Window {
                             btnSetup2.text = "button2" in pMsg ? pMsg["button2"] : ""
                             btnSetup3.text = "button3" in pMsg ? pMsg["button3"] : ""
                             btnSetup4.text = "button4" in pMsg ? pMsg["button4"] : ""
-                            view.clear()
-                            view.push(setupScreen)
+                            view.replace(setupScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "error") {
                             errorScreenLine1.text = "line1" in pMsg ? pMsg["line1"] : ""
                             errorScreenLine2.text = "line2" in pMsg ? pMsg["line2"] : ""
                             errorScreenLine3.text = "line3" in pMsg ? pMsg["line3"] : ""
-                            view.clear()
-                            view.push(errorScreen)
+                            view.replace(errorScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "settings") {
 
-                            view.clear()
-                            view.push(settingsScreen)
+                            view.replace(settingsScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "projsettings") {
                             if("line1" in pMsg) projsettingsScreenLine1.text = pMsg["line1"]
                             if("line2" in pMsg) projsettingsScreenLine2.text = pMsg["line2"]
                             if("line3" in pMsg) projsettingsScreenLine3.text = pMsg["line3"]
-                            view.clear()
-                            view.push(projsettingsScreen)
+                            view.replace(projsettingsScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "sysinfo") {
@@ -254,20 +242,17 @@ Window {
                             infoLine8.text = "line8" in pMsg ? pMsg["line8"] : ""
                             infoLine9.text = "line9" in pMsg ? pMsg["line9"] : ""
                             infoLine10.text = "line10" in pMsg ? pMsg["line10"] : ""
-                            view.clear()
-                            view.push(systemInfoScreen)
+                            view.replace(systemInfoScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "tiltmove") {
                             tiltmoveValue.text = "value" in pMsg ? pMsg["value"] : ""
-                            view.clear()
-                            view.push(tiltmoveScreen)
+                            view.replace(tiltmoveScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "towermove") {
                             towermoveValue.text = "value" in pMsg ? pMsg["value"] : ""
-                            view.clear()
-                            view.push(towermoveScreen)
+                            view.replace(towermoveScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "exception") {
@@ -275,14 +260,12 @@ Window {
                             exceptionLine2.text =  "line2" in pMsg ? pMsg["line2"] : ""
                             exceptionLine3.text =  "line3" in pMsg ? pMsg["line3"] : ""
                             exceptionQrcode.value = "qr1" in pMsg ? pMsg["qr1"] : ""
-                            view.clear()
-                            view.push(exceptionScreen)
+                            view.replace(exceptionScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "change") {
                             changeTimeExposure.text = "timeexpos" in pMsg ? pMsg["timeexpos"] : ""
-                            view.clear()
-                            view.push(changeScreen)
+                            view.replace(changeScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
                         else if(pMsg["page"] === "print") {
@@ -294,14 +277,12 @@ Window {
                             printScreenProgress.value = "progress" in pMsg ? parseInt(pMsg["progress"]) : 0
                             printScreenTimeElaps.text = "timeelaps" in pMsg ? pMsg["timeelaps"] : ""
                             printScreenTimeRemain.text = "timeremain" in pMsg ? pMsg["timeremain"] : ""
-                            view.clear()
-                            view.push(printScreen)
+                            view.replace(printScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
 
                         }
                         else if(pMsg["page"] === "homeprint") {
-                            view.clear()
-                            view.push(homeprintScreen)
+                            view.replace(homeprintScreen)
                             console.log("Seting " + pMsg["page"] + " screen.")
                         }
 
