diff --git a/setup.py b/setup.py
index 79f912b..dc4432d 100755
--- a/setup.py
+++ b/setup.py
@@ -76,7 +76,7 @@ def _dbg(s, tp=None):
 def _find_library_dirs_ldconfig():
     # Based on ctypes.util from Python 2
 
-    if sys.platform.startswith("linux") or sys.platform.startswith("gnu"):
+    if False:
         if struct.calcsize('l') == 4:
             machine = os.uname()[4] + '-32'
         else:
@@ -89,7 +89,6 @@ def _find_library_dirs_ldconfig():
             'ia64-64': 'libc6,IA-64',
             }
         abi_type = mach_map.get(machine, 'libc6')
-
         # Assuming GLIBC's ldconfig (with option -p)
         # Alpine Linux uses musl that can't print cache
         args = ['/sbin/ldconfig', '-p']
@@ -363,7 +362,7 @@ class pil_build_ext(build_ext):
         #
         # add platform directories
 
-        if self.disable_platform_guessing:
+        if self.disable_platform_guessing or True:
             pass
 
         elif sys.platform == "cygwin":
@@ -435,16 +434,6 @@ class pil_build_ext(build_ext):
 
         # FIXME: check /opt/stuff directories here?
 
-        # standard locations
-        if not self.disable_platform_guessing:
-            _add_directory(library_dirs, "/usr/local/lib")
-            _add_directory(include_dirs, "/usr/local/include")
-
-            _add_directory(library_dirs, "/usr/lib")
-            _add_directory(include_dirs, "/usr/include")
-            # alpine, at least
-            _add_directory(library_dirs, "/lib")
-
         # on Windows, look for the OpenJPEG libraries in the location that
         # the official installer puts them
         if sys.platform == "win32":
