From 4ad2c1971b3f27a18412b34726b89a25b1044da5 Mon Sep 17 00:00:00 2001
From: Roman Beranek <roman.beranek@prusa3d.com>
Date: Thu, 18 Mar 2021 07:35:43 +0100
Subject: [PATCH 2/2] axp803: add aldo1, dcdc4 and eldo2 regulators

---
 drivers/allwinner/axp/axp803.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/allwinner/axp/axp803.c b/drivers/allwinner/axp/axp803.c
index 53b11c11a..9529a1220 100644
--- a/drivers/allwinner/axp/axp803.c
+++ b/drivers/allwinner/axp/axp803.c
@@ -10,13 +10,16 @@ const uint8_t axp_chip_id = AXP803_CHIP_ID;
 const char *const axp_compatible = "x-powers,axp803";
 
 const struct axp_regulator axp_regulators[] = {
+	{"aldo1",  700, 3300, 100, NA, 0x28, 0x13, 5},
 	{"dcdc1", 1600, 3400, 100, NA, 0x20, 0x10, 0},
+	{"dcdc4",  500, 1300,  10, 70, 0x23, 0x10, 3},
 	{"dcdc5",  800, 1840,  10, 32, 0x24, 0x10, 4},
 	{"dcdc6",  600, 1520,  10, 50, 0x25, 0x10, 5},
 	{"dldo1",  700, 3300, 100, NA, 0x15, 0x12, 3},
 	{"dldo2",  700, 4200, 100, 27, 0x16, 0x12, 4},
 	{"dldo3",  700, 3300, 100, NA, 0x17, 0x12, 5},
 	{"dldo4",  700, 3300, 100, NA, 0x18, 0x12, 6},
+ 	{"eldo2",  700, 1900,  50, NA, 0x1a, 0x12, 1},
 	{"fldo1",  700, 1450,  50, NA, 0x1c, 0x13, 2},
 	{}
 };
-- 
2.31.0

