diff --git a/plat/allwinner/sun50i_a64/sunxi_power.c b/plat/allwinner/sun50i_a64/sunxi_power.c
index 07a37167..ffeeed54 100644
--- a/plat/allwinner/sun50i_a64/sunxi_power.c
+++ b/plat/allwinner/sun50i_a64/sunxi_power.c
@@ -185,6 +185,7 @@ static const struct axp_regulator {
 	unsigned char switch_reg;
 	unsigned char switch_bit;
 } regulators[] = {
+	{"aldo1",  700, 3300, 100, NO_SPLIT, 0x28, 0x13, 5},
 	{"dcdc1", 1600, 3400, 100, NO_SPLIT, 0x20, 0x10, 0},
 	{"dcdc5",  800, 1840,  10,       32, 0x24, 0x10, 4},
 	{"dcdc6",  600, 1520,  10,       50, 0x25, 0x10, 5},
