diff --git a/src/canvas.cpp b/src/canvas.cpp
index 1218015..2b91579 100644
--- a/src/canvas.cpp
+++ b/src/canvas.cpp
@@ -175,6 +175,11 @@ void Canvas::mousePressEvent(QMouseEvent* event)
     }
 }
 
+void Canvas::on_timer() {
+    yaw = fmod(yaw + 1.0, 360.0);
+    update();
+}
+
 void Canvas::mouseReleaseEvent(QMouseEvent* event)
 {
     if (event->button() == Qt::LeftButton ||
diff --git a/src/canvas.h b/src/canvas.h
index 61f627e..bc1c920 100644
--- a/src/canvas.h
+++ b/src/canvas.h
@@ -24,7 +24,7 @@ public slots:
     void set_status(const QString& s);
     void clear_status();
     void load_mesh(Mesh* m, bool is_reload);
-
+    void on_timer();
 protected:
 	void paintGL() override;
 	void initializeGL() override;
diff --git a/src/window.cpp b/src/window.cpp
index 631832c..0424d72 100644
--- a/src/window.cpp
+++ b/src/window.cpp
@@ -296,6 +296,10 @@ bool Window::load_stl(const QString& filename, bool is_reload)
         reload_action->setEnabled(true);
     }
 
+    QTimer *timer = new QTimer(this);
+    connect(timer, &QTimer::timeout, canvas, &Canvas::on_timer);
+    timer->start(10);
+
     loader->start();
     return true;
 }
