diff --git a/src/timesync/timesyncd.conf.in b/src/timesync/timesyncd.conf.in
index f91c034a0..ca2972771 100644
--- a/src/timesync/timesyncd.conf.in
+++ b/src/timesync/timesyncd.conf.in
@@ -12,8 +12,8 @@
 # See timesyncd.conf(5) for details.
 
 [Time]
-#NTP=
-#FallbackNTP={{NTP_SERVERS}}
+NTP=pool.ntp.org
+FallbackNTP=time.nist.gov time.google.com ntp.nic.cz
 #RootDistanceMaxSec=5
 #PollIntervalMinSec=32
 #PollIntervalMaxSec=2048
