commit e63b5e0dd37721ee9d6f86c0175a1fa456e36c47
Author: Filip Kotoucek <filip.kotoucek@prusa3d.cz>
Date:   Thu Nov 12 08:48:13 2020 +0100

    do not populate machine-id by default

diff --git a/systemctl b/systemctl
index 990de1a..c67293e 100755
--- a/systemctl
+++ b/systemctl
@@ -274,7 +274,7 @@ def preset_all(root):
     # For the stateless configuration, where /etc is generated at runtime
     # (for example on a tmpfs), this script shouldn't run at all and we
     # allow systemd to completely populate /etc.
-    (root / SYSCONFDIR / "machine-id").touch()
+    #(root / SYSCONFDIR / "machine-id").touch()
 
 
 def main():
