From 05d8c3f071d35aabc5b04af1602ff626a915f690 Mon Sep 17 00:00:00 2001
From: Vladimir Matena <vladimir.matena@prusa3d.cz>
Date: Fri, 13 May 2022 16:28:08 +0200
Subject: [PATCH] Drop unused mux

---
 .../dts/allwinner/sun50i-a64-prusa64-sl2.dts  | 23 -------------------
 1 file changed, 23 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts
index 31e6d05fb35e..1f8fed31cd64 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-prusa64-sl2.dts
@@ -73,29 +73,6 @@ linux,cma {
 		};
 	};
 
-	mux: mux-controller {
-		compatible = "gpio-mux";
-		#mux-control-cells = <0>;
-
-		mux-gpios = <&pio 4 12 GPIO_ACTIVE_HIGH>, /* PE12 */
-			    <&pio 6  6 GPIO_ACTIVE_HIGH>, /* PG6  */
-			    <&pio 4  6 GPIO_ACTIVE_HIGH>, /* PE6  */
-			    <&pio 3  3 GPIO_ACTIVE_HIGH>; /* PD3  */
-	};
-
-	adc-mux {
-		compatible = "io-channel-mux";
-		io-channels = <&adc>;
-		io-channels-names = "parent";
-
-		mux-controls = <&mux>;
-
-		channels = "led", "vin", "battery", "adc3", "adc4",
-			   "adc5", "adc6", "temp_ob", "dc1_current",
-			   "tenso_temp", "tenso_light", "temp3",
-			   "temp2", "temp1", "current_meas", "heater";
-	};
-
 	codec_dummy: codec-dummy {
 		#sound-dai-cells = <0>;
 		compatible = "linux,snd-soc-dummy";
