--- sl1fw	2020-02-06 16:46:47.912768062 +0100
+++ a64-fw/firmware/nginx/sl1fw	2020-02-06 12:06:20.178890889 +0100
@@ -25,10 +25,6 @@
 	server 127.0.0.1:16384;
 }
 
-upstream debugsocket {
-	server 127.0.0.1:49152;
-}
-
 # Default server configuration
 #
 server {
@@ -43,9 +39,6 @@
 	# Make site accessible from http://localhost/
 	server_name localhost;
 
-	# Require htdigest auth
-	include /etc/nginx/prusa-auth.conf;
-
 	location / {
 		try_files $uri $uri.html $uri/ =404;
 	}
@@ -62,19 +55,6 @@
 		alias /usr/share/sl1fw/multimedia;
 	}
 
-	location /dbg {
-		proxy_read_timeout 60m;
-		proxy_pass http://debugsocket;
-		proxy_http_version 1.1;
-		proxy_set_header Upgrade $http_upgrade;
-		proxy_set_header Connection $connection_upgrade;
-	}
-
-	location /logf {
-		default_type "text/plain";
-		proxy_pass http://127.0.0.1:19531/entries?_SYSTEMD_UNIT=sl1fw.service;
-	}
-
 	location /live.png {
 		sendfile on;
 		tcp_nopush on;
