diff --git a/src/update_handler.c b/src/update_handler.c
index 95ca24b..0c6806c 100644
--- a/src/update_handler.c
+++ b/src/update_handler.c
@@ -1341,6 +1341,7 @@ RaucUpdatePair updatepairs[] = {
 	{"*.squashfs", "ubivol", img_to_ubivol_handler},
 #if ENABLE_EMMC_BOOT_SUPPORT == 1
 	{"*.img", "boot-emmc", img_to_boot_emmc_handler},
+	{"*.bin", "boot-emmc", img_to_boot_emmc_handler},
 #endif
 	{"*.img", "*", img_to_raw_handler}, /* fallback */
 	{0}
