diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino.dts
index eec2a53..9cddd5e 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-olinuxino.dts
@@ -289,3 +289,22 @@
 	pinctrl-0 = <&uart3_pins>;
 	status = "okay";
 };
+
+&i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c0_pins>;
+	status = "okay";
+
+	edt-ft5x06@38 {
+		compatible = "focaltech,ft6236", "edt,edt-ft5x06";
+		reg = <0x38>;	// i2c device address
+		interrupt-parent = <&pio>;
+		interrupts = <7 7 IRQ_TYPE_EDGE_FALLING>; // PH7
+		reset-gpios = <&pio 7 8 GPIO_ACTIVE_LOW>; // PH8 GPIO_ACTIVE_LOW
+		//wake-gpios = <&pio 7 6 GPIO_ACTIVE_LOW>; // PH6
+		touchscreen-size-x = <800>;
+		touchscreen-size-y = <480>;
+		touchscreen-inverted-x;
+		touchscreen-swapped-x-y;
+	};
+};
