diff --git a/drivers/gpu/drm/sun4i/sun4i_dotclock.c b/drivers/gpu/drm/sun4i/sun4i_dotclock.c
index 417ade3d2565..3e65833aa45a 100644
--- a/drivers/gpu/drm/sun4i/sun4i_dotclock.c
+++ b/drivers/gpu/drm/sun4i/sun4i_dotclock.c
@@ -64,6 +64,9 @@ static unsigned long sun4i_dclk_recalc_rate(struct clk_hw *hw,
 	if (!val)
 		val = 1;
 
+	if (val == 4)
+		val = 12;
+
 	return parent_rate / val;
 }
 
@@ -119,7 +122,7 @@ static int sun4i_dclk_set_rate(struct clk_hw *hw, unsigned long rate,
 	u8 div = parent_rate / rate;
 
 	return regmap_update_bits(dclk->regmap, SUN4I_TCON0_DCLK_REG,
-				  GENMASK(6, 0), div);
+				  GENMASK(6, 0), 4);
 }
 
 static int sun4i_dclk_get_phase(struct clk_hw *hw)
