diff --git a/drivers/gpu/drm/panel/panel-ilitek-ili9806e.c b/drivers/gpu/drm/panel/panel-ilitek-ili9806e.c
index f3c8592ec194..14e25ca4fb9a 100644
--- a/drivers/gpu/drm/panel/panel-ilitek-ili9806e.c
+++ b/drivers/gpu/drm/panel/panel-ilitek-ili9806e.c
@@ -276,13 +276,13 @@ static int ili9806e_unprepare(struct drm_panel *panel)
 }
 
 static const struct drm_display_mode default_mode = {
-	.clock		= 19800,
-	.vrefresh	= 45,
+	.clock		= 27860,
+	.vrefresh	= 60,
 
 	.hdisplay	= 480,
-	.hsync_start	= 480 + 25,
-	.hsync_end	= 480 + 25 + 10,
-	.htotal		= 480 + 25 + 10 + 10,
+	.hsync_start	= 480 + 36,
+	.hsync_end	= 480 + 36 + 14,
+	.htotal		= 480 + 36 + 14 + 15,
 
 	.vdisplay	= 800,
 	.vsync_start	= 800 + 34,
