diff --git a/octo-uploadd/octo-uploadd.py b/octo-uploadd/octo-uploadd.py
index 3ac3b79..6bcaf19 100644
--- a/octo-uploadd/octo-uploadd.py
+++ b/octo-uploadd/octo-uploadd.py
@@ -136,8 +136,8 @@ class S(BaseHTTPRequestHandler):
 			self._set_headers()
 			self.wfile.write("<html><body><h1>Uknown address!</h1></body></html>")
         
-def run(server_class=HTTPServer, handler_class=S, port=8000):
-	server_address = ('', port)
+def run(server_class=HTTPServer, handler_class=S, port=7999):
+	server_address = ('127.0.0.1', port)
 	httpd = server_class(server_address, handler_class)
 	print 'Starting octoprint-api upload server ...'
 	httpd.serve_forever()
