From 40cafc66d38aae3eb84a6e3df674295e4eb8eb66 Mon Sep 17 00:00:00 2001
From: Roman Beranek <roman.beranek@prusa3d.com>
Date: Thu, 13 Aug 2020 11:53:48 +0200
Subject: [PATCH] disable rotation

---
 main.qml | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/main.qml b/main.qml
index 8db00ed..9799228 100644
--- a/main.qml
+++ b/main.qml
@@ -31,7 +31,7 @@ import QtQuick.VirtualKeyboard 2.3
 import QtQuick.VirtualKeyboard.Settings 2.2
 
 ApplicationWindow {
-    property bool rotate: true
+    property bool rotate: false
     id: window
     visible: true
 
@@ -421,10 +421,9 @@ ApplicationWindow {
         id: basicBottomPage
         objectName: "basicBottomPage"
         anchors.centerIn: parent
-        width: parent.height
-        height: parent.width
+        width: parent.width
+        height: parent.height
         Rectangle { anchors.fill: parent; color: "black"} // Background
-        rotation: 270
 
 
         header: Rectangle {
-- 
2.28.0

 
